app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -1097,6 +1097,15 @@ public CommCareSessionService getSession() throws SessionUnavailableException {
         }
     }
 
+
+    public UserKeyRecord getRecordForCurrentUser() {
+        try {
+            return getSession().getUserKeyRecord();
+        } catch (SessionUnavailableException e) {
+            return null;

```

Comment:
Definitely wouldn't be OK, but I guess the reason I wasn't throwing it is because it would make the code in a lot of the places that use this method very awkward. The places that this gets called are all places that we absolutely should have an active session, and if we don't, there are going to be plenty of other problems anyway, and it wouldn't even be clear what to do if the exception was caught there. This does return null which is really just another version of the same error state that the caller can check for if it makes sense to (and some of them do). Do you think it is important to throw the exception instead?